tst/com/amazon/corretto/crypto/provider/test/EvpSignatureTest.java

```diff
@@ -62,19 +71,29 @@ private static class TestParams {
         final private Signature jceVerifier;
         final private byte[] goodSignature;
 
-        public TestParams(String base, String algorithm, int length, boolean readOnly, boolean slice, KeyPair keyPair) throws GeneralSecurityException {
+        public TestParams(String base, String algorithm, int length, boolean readOnly, boolean slice,
+                KeyPair keyPair, AlgorithmParameterSpec paramSpec) throws GeneralSecurityException {
             this.base = base;
             this.algorithm = algorithm;
             this.length = length;
             this.readOnly = readOnly;
             this.slice = slice;
             this.keyPair = keyPair;
+            this.paramSpec = paramSpec;
 
             signer = getNativeSigner();
+            if (paramSpec != null) {
+                signer.setParameter(paramSpec);
+            }
             signer.initSign(keyPair.getPrivate());
             verifier = getNativeSigner();
+            verifier.setParameter(paramSpec);
```

Comment:
Why does `signer.setParameter()` need to be in an if-block but not for the `verifier`?